hermit/io.py

```diff
@@ -142,24 +140,27 @@ def read_data_from_animated_qrs(self, title: Optional[str] = None) -> Optional[s
                 progress_bar.counters.append(counter)
 
                 for c in counter:
-                    image = self.camera.get_image()
-                    mirror, data = detect_qrs_in_image(image, box_width=20)
+                    try:
+                        image = self.camera.get_image()
+                        mirror, data = detect_qrs_in_image(image, box_width=20)
+
+                        if not self.display.display_camera_image(mirror):
+                            break
+
+                        # Iterate through the identified QR codes and let the
+                        # reassembler collect them.
+                        for data_item in data:
+                            check_timer()
+                            if self.reassembler.collect(data_item):
+                                c.advance()
+
+                    except HermitError:
+                        # If for some reason we encountered an error in
+                        # scanning the qr code, parsing its contents or
+                        # consolidating it with the overall reresult,
+                        # we do not want to stop processing.
+                        pass
 
-                    if not self.display.display_camera_image(mirror):
-                        break
-
-                    # Iterate through the identified QR codes and let the
-                    # reassembler collect them.
-                    for data_item in data:
-                        check_timer()
-                        if self.reassembler.collect(data_item):
-                            c.advance()
-
-                    # await asyncio.sleep(0.05)
-
-            except InvalidQRCodeSequence as e:
```

Comment:
I'm assuming `InvalidQRCodeSequence` got replaced by `HermitError` since the latter is the actual class of the error thrown?  In this case, can we be sure that no code in L147-155 *also* raises `HermitError` ?

Reply:
`InvalidQRCodeSequence` is a subclass of `HermitError`.  There are some other places in the `for c in counter` loop that throw other `InvalicQRCodeSequence` errors. I do not think that there are any other flavors of `HermitError` being thrown here. At the moment, I would prefer to err on the side of eating too many errors in the qr scanning process, as they are probably all ephemeral. There are a few error conditions that cause us to break out of the loop - like not being able to open a display to show the feedback window. 

I had thought about catching Exception in this line, but as that is generally considered to be a bad practice I compromised by catching all of the exceptions that any of the code could be raising for things that might be recoverable problems, and went with HermitError.

Reply:
Sounds good to me.